### PR TITLE
Add Rust binding of motr

### PR DESCRIPTION
**Pull request by mengwanguc**
Add a Rust binding which supports creating, reading and writing to a motr object.

Signed-off-by: Meng Wang <mengwanguc@gmail.com>

-----
[View rendered bindings/rust/README.md](https://github.com/mengwanguc/cortx-motr/blob/rust/bindings/rust/README.md)
[View rendered doc/motr-developer-guide.md](https://github.com/mengwanguc/cortx-motr/blob/rust/doc/motr-developer-guide.md)

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: motr/client_rust.c [hunk 1]
```diff
@@ -0,0 +1,468 @@
+/* -*- C -*- */
```
Comment (huanghua78) [inline motr/client_rust.c:1]: Do you think we can move this file to "bindings/rust/" directory?

File: motr/client.h [hunk 2]
```diff
@@ -1874,6 +1874,27 @@ void m0_client_layout_free(struct m0_client_layout *layout);
 #include "motr/idx.h" /* export m0_idx operations and services to client. */
 #include "cas/cas.h"
 
+struct read_result {
```
Comment (huanghua78) [inline motr/client.h:1877]: Can we move these declarations to "bindings/rust/wrapper.h"?

File: bindings/rust/src/client.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    pub fn write_object(&self, obj_hi:u64, obj_low:u64, start:u64, len:u64, s: *mut c_char) -> std::os::raw::c_int{
+        unsafe {
+            m0_object_write(obj_hi, obj_low, start, len, s)
+        }
+    }
+}
+
+#[cfg(test)]
```
Comment (huanghua78) [inline bindings/rust/src/client.rs:81]: Do you think we can move these tests into another file, like test.rs?

File: bindings/rust/src/client.rs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+mod tests {
+    use super::*;
+    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+    static ha_addr_str:     &str = "inet:tcp:10.140.82.80@22001";
```
Comment (huanghua78) [inline bindings/rust/src/client.rs:86]: Please add comments here to explain that these parameters should be retrieved from Motr setup.
They have different values in different system.

File: bindings/rust/build.rs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+extern crate bindgen;
+extern crate run_script;
+
+use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+use run_script::ScriptOptions;
+
+fn main() {
+    println!("cargo:rustc-link-search=all=/root/cortx-motr/motr/.libs/");
```
Comment (huanghua78) [inline bindings/rust/build.rs:32]: I think we can define a variable to represent the TOP LEVEL source dir.
So, users can change it according to their own environment.

File: bindings/rust/Cargo.lock [hunk 6]
```diff
@@ -0,0 +1,477 @@
+# This file is automatically @generated by Cargo.
```
Comment (huanghua78) [inline bindings/rust/Cargo.lock:1]: This is an auto generated file.
We should not include it in source code repository.

**Comment by huanghua78** [comment]
@rkothiya , can you ask someone who are familiar with Rust to review this PR?

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**